src/rtc.rs

```diff
@@ -0,0 +1,180 @@
+/*!
+  Real time clock
+
+  A continuously running clock that counts seconds. It is part of the backup domain which means
+  that the counter is not affected by system resets or standby mode. If Vbat is connected, it is
+  not reset even if the rest of the device is powered off. This allows it to be used to wake the
+  CPU when it is in low power mode.
+
+  Since it is part of the backup domain, write access to it must be enabled before the RTC can be
+  used. See `backup_domain` for more details.
+
+  See examples/rtc.rs and examples/blinky_rtc.rs for usage examples.
+*/
+
+use stm32::{RTC};
+
+use crate::rcc::Lse;
+use crate::backup_domain::BackupDomain;
+
+use nb;
+use void::Void;
+
+
+/**
+  Interface to the real time clock
+*/
+pub struct Rtc {
+    regs: RTC,
+    _lse: Lse,
+}
+
+
+impl Rtc {
+    /**
+      Initialises the RTC. The `Lse` and `BackupDomain` structs are created by
+      `Rcc.lse.freeze()` and `Rcc.bkp.constrain()` respectively
+    */
+    pub fn rtc(regs: RTC, lse: Lse, _bkp: &BackupDomain) -> Self {
```

Comment:
Better to take a `&mut BackupDomain`

Reply:
Why? As far as I know, the backup domain can't affect the RTC apart from enabling access. I'm only taking it as a token to verify that it is enabled.

Reply:
Because in my mind, `rcc.bdcr.modify(|_, w| w.rtcen().set_bit())` should be in this method, and this is a write operation. But maybe I'm wrong?